src/frontends/tensorflow/src/translate_session.cpp

```diff
@@ -11,6 +11,35 @@
 
 using namespace ov::frontend::tensorflow;
 
+namespace {
+template <typename T>
+T reorder_ops_by_names(const std::vector<std::string>& names, const T& ops) {
+    T resulted_ops(ops.size(), nullptr);
+    if (names.size() == ops.size()) {
+        for (const auto& op : ops) {
+            const auto& op_name = op->get_friendly_name();
+            auto iter = std::find(names.begin(), names.end(), op_name);
+            if (iter != names.end()) {
+                auto ind = std::distance(names.begin(), iter);
+                if (resulted_ops[ind]) {
+                    // found two operations that are mapped to the same name
+                    // do not re-order in this case
+                    return ops;
+                }
+                resulted_ops[ind] = op;
+            } else {
+                // not found operation with the requested name
+                // do not re-order then
+                return ops;
+            }
+        }
+    } else {
+        return ops;
```

Comment:
Shouldn't this case be error and not silent skip? Same question for 3 other cases of silent return

Reply:
No, it should not because now Parameter and Result nodes re-ordering happens only for body-graphs. It was required to do for them first. For all other cases, we have to leave the legacy behavior, if names are not provided, for example, or something wrong is going.
This check will happen when we get more mature for these re-ordering to be used on the regular base.

Reply:
For main body you will have empty `names` vector and can skip reordering in that case, but if you have two vectors of different sizes this is an error and should be thrown I think. Same if you didn't find the name. Otherwise you will not know what happened if reordering didn't happen for one of those reasons.

Reply:
Added checks and raise exceptions in suspicious cases